lib/client-api/src/decorators.ts

```diff
@@ -0,0 +1,28 @@
+import { StoryContext, StoryFn, Parameters } from '@storybook/addons';
+import { ClientApiParams, DecoratorFunction, ClientApiAddons, StoryApi } from './types';
+
+const defaultContext: StoryContext = {
+  id: 'unspecified',
+  name: 'unspecified',
+  kind: 'unspecified',
+  parameters: {},
+};
+
+export const defaultDecorateStory = (storyFn: StoryFn, decorators: DecoratorFunction[]) =>
+  decorators.reduce(
+    (decorated, decorator) => (context: StoryContext = defaultContext) =>
+      decorator(
+        // You cannot override the parameters key, it is fixed
+        ({ parameters, ...innerContext }: StoryContext) =>
```

Comment:
@ndelangen are there *any* keys that we want to pass through in the context?

It occurs to me that you shouldn't be able to override things like `id`, `name` or `kind` either?

Reply:
Yeah merging things into the StoryContext is pretty dangerous, it can be useful too...

Reply:
Which things do you think we should allow people to override>

Reply:
I agree. I think we should keep a list of properties we don't allow maybe?

Reply:
Well I mean I am not even sure what would be in a potential whitelist.

Here is what we currently put on the context:

`...identification`: (I don't think we should allow overriding any of these):
 - `id`,
 - `kind`,
 - `name`,
 - `story` == `name` (<- legacy, we can probably drop this in 6.0)

`hooks` - not currently overridable, nor should it be
`parameters` - was overridable in 5.x, no longer is in 6.0
`args` (<- not in this PR, but coming, also not overridable)

That is it. So currently all we would allow is for the user to add arbitrary keys *outside* of that list to the context. I guess maybe that is useful? You could see an addon doing that I guess. It's not currently a recommended approach but we could add it.

Reply:
Or maybe addons are already adding arbtirary keys -- do you know?